public/app/features/dashboard/components/DashExportModal/DashboardExporter.ts

```diff
@@ -116,7 +116,7 @@ export class DashboardExporter {
               pluginName: ds.meta?.name,
             };
 
-            if (obj.datasource === null || typeof obj.datasource === 'string') {
+            if (!obj.datasource || typeof obj.datasource === 'string') {
               obj.datasource = '${' + refName + '}';
```

Comment:
How come this part isn't using the new data source ref object?

Reply:
see josh's comment as well 😄